convert2rhel/systeminfo.py

```diff
@@ -152,49 +155,74 @@ def get_system_release_file_content():
+            # - "release ", parsed as (?:release\s)?,
+            #   is a literal string. It is optional, and it is dropped when parsed, so this group doesn't have a name.
+            # - <full_version>, parsed as a named group '(?P<full_version>[.\d]+)',
+            #   is a string of arbitrary length containing numbers and dots, for example 8.1.1911 or 7.9.
+            #   (For now we do not have examples with version strings containing letters or other symbols)
+            # - <Beta>, just the word Beta which may appear in some cases, parsed as '(?:\sBeta)?'.
+            # - <dist_id>, parsed as a named group (?P<dist_id>.+), is optional and when it is present, it must appear
+            #   in brackets. Thus, the named group is nested under an unnamed group which starts from a space and then
+            #   has brackets '(\s\( ...here goes the nested group... \))?'
+            #
+            # Example:
+            #
+            #   CentOS Stream release 8
+            #   <    name   > <      ><full_version>
+            #
+            #   CentOS Linux release 8.1.1911        (Core     )
+            #   <    name  > <      ><full_version><  <dist_id>>
+            #   CentOS Linux release 8.1.1911      Beta       (Core     )
+            #   <    name  > <      ><full_version><    ><  <dist_id>>
+            r"^(?P<name>.+?)\s(?:release\s)?(?P<full_version>[.\d]+)(?:\sBeta)?(\s\((?P<dist_id>.+)\))?$",
+            content,
+        )
 
-    def _get_system_distribution_id(self, system_release_content=None):
-        """Return the distribution id from the system release file.
+        if not match:
+            self.logger.critical_no_exit("Couldn't parse the system release content string: %s" % content)
+            return {}
 
-            .. note::
-                This distribution id differs from the property `id` we have in the SystemInfo class
-                as this id is the last thing that appears on the system-release file as noted by
-                the example below.
+        name = match.group("name")
+        system_id = name.split()[0].lower()
 
-        Examples:
-        Oracle Linux Server release 7.8      <- None
-        CentOS Linux release 7.6.1810 (Core) <- Core
-        CentOS Linux release 8.1.1911 (Core) <- Core
+        distribution_id = match.group("dist_id")
 
-        :returns: The distribution id from the system release file if any.
-        :rtype: str | None
-        """
-        content = self.system_release_file_content if not system_release_content else system_release_content
-        match = re.search(r"(?<=\()[^)]*(?=\))", content)
-        if not match:
-            return None
+        full_version = match.group("full_version")
+        version_numbers = full_version.split(".")
+        major = int(version_numbers[0])
+        if len(version_numbers) > 1:
+            minor = int(version_numbers[1])
+        else:
+            minor = 10  # Stream 8 EOL version maps to 8.10.
```

Comment:
In case of CentOS Stream we do not have version_numbers[1] at all. We could put "None", but that would lead to cascading changes further in the code and I wanted to keep the change compatible with the current workflow.

10 is the "largest RHEL minor version ever released". By nature of the CentOS Stream, it runs ahead of the current RHEL minor release for its entire lifecycle, so CentOS Stream is always newer than 8.4 or 8.5 or 8.8. Fully updated CentOS Stream 8 must have higher version number than any released RHEL 8 version, all the way until RHEL 8.10. At RHEL 8.10 release RHEL catches up while CentOS Stream freezes, and that freezed state is .10.

Reply:
What about all RHEL-likes though? Alma, Rocky etc. I'm just worried that this is cut short by only thinking about this specific scenario of CentOS Stream conversions

Is it guaranteed to only be set to 10 when we cannot find the minor version?

Reply:
Or I would add better comment describing what's going on there. And in case it would cause issues in the feature, it will be at least easy to find where is the problem

Reply:
I have updated the PR with a more detailed comment.